lnwallet/channel.go

```diff
@@ -6033,6 +6033,13 @@ func (lc *LightningChannel) availableCommitmentBalance(view *htlcView) (
 		case ourBalance >= htlcCommitFee+nonDustHtlcAmt:
 			ourBalance -= htlcCommitFee
 
+		// We have enough to pay the non-dust HTLC, but not enough to
+		// cover the resulting fee. Therefore we set our available
+		// balance slightly below the non-dust amount to allow HTLCs up
+		// to that size.
+		case ourBalance >= baseCommitFee+nonDustHtlcAmt:
+			ourBalance = nonDustHtlcAmt - 1
```

Comment:
What's the rationale for the minus one here? Also why aren't we _subtracting_ the base fee _plus_ the amount of funds that need to enter limbo for the HTLC?

Reply:
If we send `>= nonDustHtlcAmt` we will need to pay the extra htlc fee, which we cannot afford.

`nonDustHtlcAmt-1` is the largest amount we can send without having to pay this fee.

Reply:
I did split them with the intention that the added case goes together with the testcase and commit message to explain what it achieves.